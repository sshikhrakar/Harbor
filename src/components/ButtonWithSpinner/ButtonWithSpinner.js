import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  ActivityIndicator,
} from 'react-native';
import PropTypes from 'prop-types';
import colors, { DEFAULT_BUTTON_OPACITY } from '../../config/colors';
import fonts from '../../config/fonts';

/**
 * Button with spinner.
 *
 * @param {String} text
 * @param {String} onPress
 *
 * @returns {jsx}
 */
function ButtonWithSpinner({ text, onPress, isLoading }) {
  return (
    <View style={ styles.buttonContainer }>
      <TouchableOpacity
        style={ styles.button }
        onPress= { onPress }
        activeOpacity={ DEFAULT_BUTTON_OPACITY }
      >
        {
          !isLoading ?
            <Text style={ styles.buttonText }>
              { text }
            </Text>
            :
            <ActivityIndicator animating={ isLoading } color={'white'} />
        }
      </TouchableOpacity>
    </View>
  );
}

ButtonWithSpinner.propTypes = {
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  buttonContainer: {
    flex: 1,
    paddingVertical: 10,
  },

  button: {
    flex: 0.8,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.ASBESTOS,
  },

  buttonText: {
    fontSize: 20,
    marginVertical: 5,
    color: colors.SILVER,
    fontFamily: fonts.primary.LIGHT,
  },
});

export default ButtonWithSpinner;
