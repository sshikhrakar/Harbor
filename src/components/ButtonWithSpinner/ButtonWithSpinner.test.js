import React from 'react';
import renderer from 'react-test-renderer';

import ButtonWithSpinner from '../ButtonWithSpinner';

describe('COMPONENTS: ButtonWithSpinner', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <ButtonWithSpinner text="TestButtonText" onPress={ () => { console.log('Button works!'); } }/> // eslint-disable-line no-console
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
