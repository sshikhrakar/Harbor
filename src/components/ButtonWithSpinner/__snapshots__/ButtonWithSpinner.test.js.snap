// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS: ButtonWithSpinner renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingVertical": 10,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#7f8c8d",
        "flex": 0.8,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ecf0f1",
          "fontFamily": "Lato-Light",
          "fontSize": 20,
          "marginVertical": 5,
        }
      }
    >
      TestButtonText
    </Text>
  </View>
</View>
`;
